cmd/packer-plugin-check/main.go

```diff
@@ -129,11 +129,12 @@ func discoverAndLoad() error {
 	}
 
 	// TODO: validate correctness of plugins loaded by checking them against the output of the `describe` command.
-	builders, provisioners, postProcessors := config.GetPlugins()
+	builders, provisioners, postProcessors, datasources := config.GetPlugins()
 	if len(builders) == 0 &&
 		len(provisioners) == 0 &&
-		len(postProcessors) == 0 {
-		return fmt.Errorf("couldn't load any Builder/Provisioner/Post-Processor from the plugin binary")
+		len(postProcessors) == 0 &&
+		len(datasources) == 0 {
+		return fmt.Errorf("couldn't load any Builder/Provisioner/Post-Processor/Datasource from the plugin binary")
```

Comment:
These fields are part of config, I think we can simply remove the `GetPlugins` call and make those public. ( Actually I made them public for #10304 because I plan on using that config struct everywhere 🙂  )